partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,272 @@
+ *
+ */
+
+/**
+ * Jetpack Meta Data for WHMCS module.
+ * @return array
+ */
+function jetpack_MetaData()
+{
+    return array(
+        'DisplayName' => 'Jetpack by Automattic',
+        'Description' => 'Use this module to provision Jetpack plans with your Jetpack hosting partner account',
+        'APIVersion' => '1.1',
+        'RequiresServer' => false,
+    );
+}
+
+
+/**
+ * Basic configuration options required for a partner to get
+ * a Jetpack plan provisioned. Currently a partner client id
+ * and secret are the only host partner options needed to get
+ * an access token to provision a Jetpack plan
+ * @return array
+ */
+function jetpack_ConfigOptions()
+{
+    return array(
+        'Jetpack Partner Client ID' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        ),
+        'Jetpack Partner Client Secret' => array(
+            'Type' => 'text',
+            'Size' => '256',
+        )
+    );
+}
+
+
+/**
+ * Equivalent to /provision. Create a Jetpack plan using
+ * a Jetpack Hosting partner account.
+ *
+ *
+ * @param array $params
+ * @return string 'success'
+ * @throws Exception
+
+ */
+function jetpack_CreateAccount(array $params)
+{
+    try {
+        validate_required_fields($params);
+
+        $access_token = get_access_token($params);
+        $provisioning_url = "https://public-api.wordpress.com/rest/v1.3/jpphp/provision";
+        $request_data = array (
+            'plan' => strtolower($params['customfields']['Plan']),
+            'siteurl' => $params['customfields']['Site URL'],
+            'local_user' => $params['customfields']['Local User'],
+            'force_register' => true,
+        );
+
+        $response = make_api_request($provisioning_url, $access_token, $request_data);
+        if ($response->success && $response->success == true) {
+            if ($response->next_url) {
+                save_provisioning_details($response->next_url, $params);
+            } else if(!$response->next_url && $response->auth_required)  {
+                save_provisioning_details($response->next_url, $params, true);
+            }
+
+            return 'success';
+        }
+    } catch(Exception $e) {
+        logModuleCall('jetpack', __FUNCTION__, $params, $e->getMessage(), $e->getMessage());
+        return $e->getMessage();
+    }
+
+}
+
+/**
+ * Equivalent to partner/cancel. Cancel a Jetpack plan using
+ * using a Jetpack Hosting partner account.
+ *
+ * @param array $params
+ * @return string
+ * @throws Exception
+ */
+function jetpack_TerminateAccount(array $params)
+{
+    try {
+        validate_required_fields($params);
+
+        $access_token = get_access_token($params);
+        $clean_url = str_replace('/', '::', $params['customfields']['Site URL']);
+        $url = 'https://public-api.wordpress.com/rest/v1.3/jpphp/' . $clean_url .'/partner-cancel';
+        $response = make_api_request($url, $access_token);
+        if ($response->success == true) {
```

Comment:
It's possible for `$response->success` to be false when cancelling, which specifically happens when there was not a plan to cancel. For example, if a hosting partner calls the `/parter-cancel` endpoint twice, the first time `success` should be true and the second time `success` would be false.

Reply:
@ebinnion good catch and thanks! I was previously handling anything that wasn't 2xx in the function that made the requests and didn't missed this. I've updated the function to now specifically handle where the response is a 200 and success is false